### ISSUE TITLE
Node.js Technical Steering Committee (TSC) Meeting 2020-04-16

### ISSUE THREAD

**Issue by mhdawson** [nodejs/TSC#848]
## Time

**UTC Thu 16-Apr-2020 15:00 (03:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Thu 16-Apr-2020 08:00 (08:00 AM) |
| US / Mountain | Thu 16-Apr-2020 09:00 (09:00 AM) |
| US / Central  | Thu 16-Apr-2020 10:00 (10:00 AM) |
| US / Eastern  | Thu 16-Apr-2020 11:00 (11:00 AM) |
| London        | Thu 16-Apr-2020 16:00 (04:00 PM) |
| Amsterdam     | Thu 16-Apr-2020 17:00 (05:00 PM) |
| Moscow        | Thu 16-Apr-2020 18:00 (06:00 PM) |
| Chennai       | Thu 16-Apr-2020 20:30 (08:30 PM) |
| Hangzhou      | Thu 16-Apr-2020 23:00 (11:00 PM) |
| Tokyo         | Fri 17-Apr-2020 00:00 (12:00 AM) |
| Sydney        | Fri 17-Apr-2020 01:00 (01:00 AM) |

Or in your local time:
* http://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Technical%20Steering%20Committee%20(TSC)+Meeting+2020-04-16&iso=20200416T15
* or http://www.wolframalpha.com/input/?i=03PM+UTC%2C+Apr+16%2C+2020+in+local+time

## Links

* Minutes Google Doc: <https://docs.google.com/document/d/1AR3CTuYtKRCrVEOCubq444jicanlt8XfRlFvPc5_bqQ/edit>

## Agenda

Extracted from **tsc-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/node

* Nominating @puzpuzpuz as a collaborator [#32555](https://github.com/nodejs/node/issues/32555)
* Nominating @juanarbol as a collaborator [#32621](https://github.com/nodejs/node/issues/32621)  

### nodejs/docker-node

* Suggestion for dropping Yarn from Node 14 release [#1238](https://github.com/nodejs/docker-node/issues/1238)

### nodejs/TSC

* TSC Chair Election 2020 - proposed schedule [849](https://github.com/nodejs/TSC/issues/849)
* Node.js future directions - any interest in online or in person summit? [#797](https://github.com/nodejs/TSC/issues/797)

### nodejs/admin

* Install github dco app for use in ci-config-travis and ci-config-github-actions repos [#492](https://github.com/nodejs/admin/issues/492)

## Invited

* Anna Henningsen @addaleax (TSC)
* Anatoli Papirovski @apapirovski (TSC)
* Beth Griggs @BethGriggs (TSC)
* Сковорода Никита Андреевич @ChALkeR (TSC)
* Colin Ihrig @cjihrig (TSC)
* Daniel Bevenius @danbev (TSC)
* Franziska Hinkelmann @fhinkel (TSC)
* Gabriel Schulhof @gabrielschulhof (TSC)
* Gireesh Punathil @gireeshpunathil (TSC)
* James Snell @jasnell (TSC)
* Joyee Cheung @joyeecheung (TSC)
* Matheus Marchini @mmarchini (TSC)
* Matteo Collina @mcollina (TSC)
* Michael Dawson @mhdawson (TSC)
* Myles Borins @MylesBorins (TSC)
* Ruben Bridgewater @BridgeAR (TSC)
* Sam Roberts @sam-github (TSC)
* Michaël Zasso @targos (TSC)
* Tobias Nießen @tniessen (TSC)
* Rich Trott @Trott (TSC)
* Shelley Vohr @codebytere (TSC)

## Notes

The agenda comes from issues labelled with `tsc-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

Zoom link: https://zoom.us/j/611357642
Regular password

## Public participation

We stream our conference call straight to YouTube so anyone can listen to it live, it should start playing at **https://www.youtube.com/c/nodejs+foundation/live** when we turn it on. There's usually a short cat-herding time at the start of the meeting and then occasionally we have some quick private business to attend to before we can start recording & streaming. So be patient and it should show up.

Many of us will be on IRC in #node-dev on Freenode if you'd like to interact, we have a Q/A session scheduled at the end of the meeting if you'd like us to discuss anything in particular. @nodejs/collaborators in particular if there's anything you need from the TSC that's not worth putting on as a separate agenda item, this is a good place for that.

---

**Invitees**

Please use the following emoji reactions in this post to indicate your
availability.

* :+1: - Attending
* :-1: - Not attending
* :confused: - Not sure yet

**Comment by Trott** [comment]
Who on the TSC (if anyone) is involved in either our Docker releases or yarn? Trying to get a good grip on https://github.com/nodejs/docker-node/issues/1238 which is on the agenda this week. Maybe the people who are releasers will have informed opinions?

**Comment by sam-github** [comment]
That's one of those cases where its unlikely the TSC is going to swoop in and tell the docker-node team what to do, but we can discuss in the TSC meeting, and they might find some our opinions helpful in breaking any impasse. I have a bit of contact with the docker-node team when we coordinate security releases, but many of use _use_ docker, so might have opinions on what is in the image. Its easy to come up with ideas for things other people could do, though, it will be the docker-node team that has to decide how much work they have the capacity to commit to.

**Comment by mhdawson** [comment]
I've had contact with the Docker team for past security releases and when we helped make them multi-arch but I'm not involved (and am not aware of any other TSC members who are) in the day to day work.  As @sam-github some discussion/opinions from the TSC may help the Docker team decide what they'd like to do.

**Comment by mhdawson** [comment]
There is not a lot on the agenda, but we do have the docker issue to discuss and we've skipped the strategic initiatives for the last number of meetings so I think it would still be good to get together and try to go through those.  It might be shorter than the full hour, but I think that would be ok.

**Comment by mhdawson** [comment]
Adding https://github.com/nodejs/TSC/issues/849 to the agenda so that we are ready to open the call for nominations April 16 unless there are concerns.

**Comment by Trott** [comment]
> we've skipped the strategic initiatives for the last number of meetings

We really should not do those during the meeting. We should do those in the issue tracker. We can then discuss anything significant but for the most part, it's really not a good use of meeting time in my opinion. (Unpopular opinion?)

**Comment by mhdawson** [comment]
@Trott I personally don't agree with your comment about the strategic initiatives as they don't seem to happen at all otherwise and we often end up sharing interesting info on things like v8 currency and others I don't keep up with day to day (at least to me).  Just my opinion so it depends on what the majority of the TSC members think.

**Comment by mhdawson** [comment]
For awareness adding https://github.com/nodejs/admin/issues/492 to see if there are any concerns.

**Comment by gireeshpunathil** [comment]
+1 to retaining strategic initiative updates in TSC. Those are key items, and this is the only forum through which I get an update on the latest on those.

**Comment by Trott** [comment]
> @Trott I personally don't agree with your comment about the strategic initiatives as they don't seem to happen at all otherwise and we often end up sharing interesting info on things like v8 currency and others I don't keep up with day to day (at least to me). Just my opinion so it depends on what the majority of the TSC members think.

I would prefer that people provide the updates here, in the issue for the meeting, and then they can be reviewed at the meeting. That solves a bunch of issues:

* Strategic initiatives will either take less time at the meeting or be more focused. Or maybe both.
* We get updates on initiatives even if the people who are involved in them can't make the meeting that particular week.
* It provides a lower-friction medium to share the information. (Well, I guess that depends how you define "friction" but that's my opinion anyway.)

That said, for that to happen will require agreement and prompting. I suppose I could stop advocating for this and just start prompting if I felt that strongly....

**Comment by Trott** [comment]
Strategic initiatives!

@MylesBorins Any updates on Modules this week?

@mcollina Anything on core promises APIs?

@targos Anything on V8 currency?

@jasnell Anything on QUIC?

@joyeecheung Anything on startup performance? (I imagine that PR opened by @addaleax might be worth mentioning?)

@mhdawson Build resources! Anything to report this week?

**Comment by Trott** [comment]
Moderation team report! This is my draft. If anyone on the team thinks it's not complete/accurate, I'll post again.

Nothing significant to report this week. Some spam comments dealt with and associated users blocked. That's about it.

@nodejs/tsc @nodejs/community-committee @nodejs/moderation

**Comment by jasnell** [comment]
re: QUIC ... work on the implementation is continuing. The OpenSSL patch did not require any updates for OpenSSL 1.1.1f. We'll see what happens with the upcoming 1.1.1g. As a backup plan, I have started investigating what it would take to implement the QUIC support as a native addon and the work is absolutely *not* trivial. The implementation currently relies on a significant part of Node.js core internals that are not exposed via N-API so the ideal path is still landing in core.

**Comment by Trott** [comment]
> Moderation team report! This is my draft. If anyone on the team thinks it's not complete/accurate, I'll post again.
> 
> Nothing significant to report this week. Some spam comments dealt with and associated users blocked. That's about it.
> 
> @nodejs/tsc @nodejs/community-committee @nodejs/moderation

I forgot to also mention that during the downloads outage, @bitandbang locked a few issues since there were duplicates across repositories and there were comments in the issues directing people to the issue in the build repo.

**Comment by targos** [comment]
V8 currency:

- V8 8.1 is now stable. v14.0.0 will go out with it as planned.
- V8 8.3 will enter beta tomorrow. @mmarchini is working on the update for master [here](https://github.com/nodejs/node/pull/32831).
  - There's an open question about ABI-breaking changes in it: do we want to include some of them in v14.0.0 to make it easier to upgrade V8 in a future minor release of v14.x? We only have a few days left to do that.
- Canary is in a relatively good shape. We've got one test failure to investigate (related to WeakRefs, an experimental JS feature): https://github.com/nodejs/node-v8/issues/156

**Comment by bnoordhuis** [comment]
> implement the QUIC support as a native addon and the work is absolutely not trivial. The implementation currently relies on a significant part of Node.js core internals that are not exposed via N-API

@jasnell I'd be interested to hear more about that. How about we move that to a separate issue?

**Comment by cjihrig** [comment]
^ another benefit to doing strategic initiatives on the issue tracker is that it's easier for non-TSC members to get involved in the discussion. Maybe that is what @Trott meant by a "lower-friction medium to share the information."

**Comment by jasnell** [comment]
@bnoordhuis ... we can discuss over in the nodejs/quic repo issue tracker ... https://github.com/nodejs/quic/issues/371

**Comment by jasnell** [comment]
@mhdawson ... the QUIC PRs can be taken off the agenda for this week. I will remove the tsc-agenda label from those.

**Comment by mhdawson** [comment]
@cjihrig I would agree with the benefits of using the issue tracker, but despite encouragement from Rich for updates outside the meeting they did not occur regularly or at all, so my take is that we just wont' get them at all....

**Comment by mhdawson** [comment]
I see that @Trott is trying to get people to comment in the issue for the strategic initiatives, if we get people doing that consistently then I'm less worried. We could take the approach of

1) Ask people to comment in the issue
2) In the meeting for those that have been covered simply ask if there are any outstanding questions
3) For those that have not been covered ask for a quick update from the Champion if they are there.

That might get more of the discussion into a github issue without risking not getting any update/discussion at all. @Trott thoughts ?

**Comment by mhdawson** [comment]
Build resources update
* Based on the request to the Board for help find build resources we've had a few new volunteers
  raise their hand from Google and GitHub based on the encouragement from their Board Directors.

* At the same time options for hiring resources that would help resolve key, urgent problems that 
  get in the way of collaborator progress are being explored in parallel as it was/is not clear if we can 
  get  enough new volunteers with enough time to address the resource issues we've raised.  I've 
  gotten some feedback from the most active build WG members to scope what we might do initially
  and will work with resources proposed by the Foundation to see if the cost is something that might 
  be reasonable.

**Comment by MylesBorins** [comment]
Modules update

* Aiming to remove warning for Node.js 14 (likely after 14.0.0 due to timing)
* Aiming to remove some of the current warnings on 12 in an upcoming release
* Aiming to remove the flag in 12.17.0
* Ecosystem adoption of new features is growing and biggest issue for adoption at the moment appears to be flags + warning not behavior or desire to adopt

**Comment by tniessen** [comment]
I might be late or not attend at all today, sorry.

**Comment by ErickWendel** [comment]
Watching, it's being nice!

**Comment by mhdawson** [comment]
@ErickWendel thanks :).

PR for minutes: https://github.com/nodejs/TSC/pull/850
